### DISCUSSION TITLE
[BUG] New-PnPUPABulkImportJob not working after registering pnp.powershell as new app for tenant

### DISCUSSION THREAD

**Discussion by Id10tSupport** [pnp/powershell#4269] [General]
We recently created the new app registration for PnP.PowerShell in our tenant and now New-PnPUPABulkImportJob is no longer working.  It was working prior to the deprecation of the multi-tenant app registration.

### Example of script used:
$FileURL = "https://xxxxxx.sharepoint.com/sites/search/Bio/res2024_09_10.json"

#Update About Me for accounts in JSON file
$propertyMap = New-Object -type 'System.Collections.Generic.Dictionary[String,String]'
$propertyMap.Add("EmpID","EID")
$propertyMap.Add("About Me","AboutMe")
$propertyMap.Add("Quick Dial","QuickDial")
$propertyMap.Add("Assistant Name","AssistantName")
$propertyMap.Add("Assistant Phone","AssistantPhone")
$propertyMap.Add("AssistantEmail","AssistantEmail")
$propertyMap.Add("BusinessUnit","BusinessUnit")
$propertyMap.Add("Floor","Floor")
$propertyMap.Add("CellPhone","CellPhone")

New-PnPUPABulkImportJob -Url $FileURL -IdType PrincipalName -IdProperty "OperatingSystemID" -UserProfilePropertyMapping $propertyMap

### Response received
New-PnPUPABulkImportJob : Access Denied: This application does not have the required permissions to access profile information.

I have verified that the app does have User.ReadWrite.All access granted in SharePointOnline for User's Profiles as well as Microsoft Graph for User's Profiles

### What is the version of the Cmdlet module you are running?
PSVersion                      5.1.14409.2001 
PnP.PowerShell             1.12.4

### Which operating system/environment are you running PnP PowerShell on?
Windows Server 2012 R2

It should be noted that I get the same error response on a Windows 2022 Server using PSVersion 7.4.4 and PnP.PowerShell 2.1.1

**Comment by gautamdsheth** [comment]
You also need to give SharePoint > Sites.FullControl or Sites.Manage permissions as it creates a file in SharePoint site.

Check the docs for that here:

https://pnp.github.io/powershell/cmdlets/New-PnPUPABulkImportJob.html

Am moving it to a discussion, as it is a configuration issue and not a cmdlet issue.

**Reply by Id10tSupport** [reply]
The app has already been granted the following:
AllSites.FullControl	Delegated	Have full control of all site collections
Sites.FullControl.All	Application	Have full control of all site collections

Is that not enough?

**Reply by Id10tSupport** [reply]
as well as having User.ReadWrite.All

**Reply by gautamdsheth** [reply]
How are you connecting to SharePoint in connect-pnponline ? 

I think this is step is also required as far as I remember

https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/bulk-user-profile-update-api-for-sharepoint-online#frequently-asked-questions

**Reply by Id10tSupport** [reply]
Connect-PnPOnline -Url $SiteURL -ClientID $ClientID -Credential $Cred

**Reply by Id10tSupport** [reply]
prior to having to add the ClientID to the connect action, I never had to do anything other than use the bulk import cmdlet

**Reply by gautamdsheth** [reply]
Could it be this ? 

<img width="466" alt="image" src="https://github.com/user-attachments/assets/03366c34-aadc-481b-a0b2-312924e61fb6">

**Reply by Id10tSupport** [reply]
when was this change made?  Is that a new limitation?

**Reply by gautamdsheth** [reply]
Hmm no idea, just saw this in the docs in the link shared above. Also, maybe check that the user has access to the site collection where you are storing the JSON as well as SharePoint admin site.

**Reply by Id10tSupport** [reply]
Account has Full Control access to the site and is a SharePoint Admin.  Like I said, this was working right up until the recent multi-tenant app registration was replaced with single tenant.  The only change I was expecting to have to make in the automation scripts was to add the ClientID information to the Connect-PnPOnline.

**Reply by gautamdsheth** [reply]
https://pkbullock.com/blog/2024/permissions-for-pnp-management-shell-app

These were all the permissions that were requested by the now deleted app, maybe you need to add termstore.readwrite.all as well in case you are setting some terms in the profile property.
